### PR TITLE
Refactor sha[256/512]crypt to make code cleaner

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/opencl_sha256crypt_fmt_plug.c [hunk 1]
```diff
@@ -586,10 +587,9 @@ static int crypt_all(int *pcount, struct db_salt *_salt)
 		BENCH_CLERROR(clEnqueueNDRangeKernel(queue[gpu_id], preproc_kernel, 1,
 		                                     NULL, &gws, lws, 0, NULL, multi_profilingEvent[4]),
 		              "failed in clEnqueueNDRangeKernel II");
+		iterations = ocl_autotune_running ? 3 : (salt->rounds / HASH_LOOPS);
```
Comment (solardiz) [inline src/opencl_sha256crypt_fmt_plug.c:591]: I think we can move `i` and `iterations` to this scope, and use `unsigned int i, iterations  = ocl_autotune_running ? 3 : (salt->rounds / HASH_LOOPS);` here.

As the code is currently written, I expect some compilers with some options would warn us we got "comparison between signed and unsigned".

**Review by solardiz — APPROVED** [review]
Looks good to me, but I suggested a change.

File: src/opencl_sha256crypt_fmt_plug.c (hunk 1 above)
Comment (ghost) [reply src/opencl_sha256crypt_fmt_plug.c:591]: To avoid the "forbids mixed declarations and code", I added curly brackets (without indentation).

Comment (magnumripper) [reply src/opencl_sha256crypt_fmt_plug.c:591]: I think we stopped forbidding that some time ago?

Comment (ghost) [reply src/opencl_sha256crypt_fmt_plug.c:591]: Ok, but should I add the braces (and avoid mixing) or not?

Comment (solardiz) [reply src/opencl_sha256crypt_fmt_plug.c:591]: @claudioandre-br magnum is correct, we no longer avoid mixed declarations and code. Please drop the extra curly braces. Thanks!

**Pull request merged**